### DISCUSSION TITLE
CPU and GPU fan speed lock

### DISCUSSION THREAD

**Discussion by joaocarloscr** [seerge/g-helper#2000] [Q&A]
Is there a way to "lock" the speeds of the fans to the speed of the faster one?
I have a custom curve but the CPU heats up a lot more, so the fan ends up going 100%, while the GPU fan stays at 40%. 
I would love to "lock" the GPU fan speed, so that they run at the same speeds, allowing the heat to dissipate more efficiently from the chassis, while not needing one of them at max speed.

I have a 2021 G14 - R9 5900HS / RTX 3060 and I'm using the latest version of G-Helper. 
Thanks in advance!

**Comment by joaocarloscr** [comment]
Also, while using the default Turbo fan curve, the fans will always run at the same speed. But they won't get to max speeds and the CPU will be at 94ºC+.
If I set the speed from both fans manually to 100%, the CPU will be at a much lower temperature.

**Comment by seerge** [comment]
@joaocarloscr hello, there is no way to "lock" fans, and control them in realtime anyhow. Fans are always controlled by BIOS, and if they behave like that - it's how they are programmed :)

You can try to experiment with custom fan curves , or just uncheck "apply" and stick to defaults

**Comment by joaocarloscr** [comment]
The thing is, the default does try to make them run at the same speeds, even with the CPU producing all the heat, but the default curve will let the CPU reach 94ºC+ never maxing out the fans.
So I need a custom curve. 
But with a custom curve, the CPU fan goes to 100%, the GPU one stays at 40%, and the temperatures are the same because the GPU fan isn't helping. 
The only way to lower temps is set them to 100% manually.

Are there plans in the future to make the GPU fan follow the CPU fan speed?

Thanks for the reply and thanks for G-Helper. It's an awesome tool.

**Reply by seerge** [reply]
@joaocarloscr there are no such plans, as it's impossible to control the fans from windows :) It's possible to set a custom fan curve in temperature - fan% points, or not to set it (and stick to whatever default is). That's it.

You can try to edit fan curve, for example shift first dot in it higher (i think it was some models that were changing fan behavior based on that).

**Reply by joaocarloscr** [reply]
I understand. I will try to make a higher curve for the GPU.
And again, thanks a lot for G-Helper, would pay for it if it wasn't free :)

**Comment by erfanHemayati** [comment]
Hi, 
first of all, I want to thank you for your efforts on this app; it is a lifesaver.
I was thinking about such a thing that @joaocarloscr mentioned, as the default profile runs my laptop a little warmer than I like. I was wondering if it is possible to update the fan curve automatically in the background based on the average temp of the CPU. For example, if the CPU is hot for a while then a higher fan curve is made and applied for the GPU in the background. Subsequently, the BIOS reads and adopts the new fan curve. Do you think that's possible to implement?
